File: app/src/main/java/com/alienstar/cyrus/advancedandroid/di/Injector.java
package com.alienstar.cyrus.advancedandroid.di;

import android.app.Activity;

import com.alienstar.cyrus.advancedandroid.base.BaseActivity;
import com.bluelinelabs.conductor.Controller;

/**
 * Created by cyrus on 3/5/18.
 */

public class Injector {
    private Injector(){

    }
    public static void inject(Activity activity){
        ActivityInjector.get(activity).inject(activity);

    }
    public static void clearComponent(Activity activity){
        ActivityInjector.get(activity).clear(activity);
    }

    public static void inject(Controller controller){
        ScreenInjector.get(controller.getActivity()).inject(controller);
    }

    public static void clearComponent(Controller controller){
        ScreenInjector.get(controller.getActivity()).clear(controller);
    }
}

File: app/src/main/java/com/alienstar/cyrus/advancedandroid/di/Injector.kt
package com.alienstar.cyrus.advancedandroid.di

import android.app.Activity

import com.alienstar.cyrus.advancedandroid.base.BaseActivity
import com.bluelinelabs.conductor.Controller

/**
 * Created by cyrus on 3/5/18.
 */

object Injector {
    fun inject(activity: Activity) {
        ActivityInjector.get(activity).inject(activity)

    }

    fun clearComponent(activity: Activity) {
        ActivityInjector.get(activity).clear(activity)
    }

    fun inject(controller: Controller) {
        ScreenInjector.get(controller.activity).inject(controller)
    }

    fun clearComponent(controller: Controller) {
        ScreenInjector.get(controller.activity).clear(controller)
    }
}
